pkg/url/url.go

```diff
@@ -30,6 +30,45 @@ func (urls URLList) Get(urlName string) URL {
 
 }
 
+// Get returns URL definition for given URL name
+func Get(client *occlient.Client, localConfig *config.LocalConfigInfo, urlName string, applicationName string) (URL, error) {
+	remoteUrlName, err := util.NamespaceOpenShiftObject(urlName, applicationName)
+	if err != nil {
+		return URL{}, errors.Wrapf(err, "unable to create namespaced name")
+	}
+
+	// Check whethter remote already created the route
+	remoteExist := true
+	route, err := client.GetRoute(remoteUrlName)
+	if err != nil {
+		remoteExist = false
+	}
+
+	localConfigURLs := localConfig.GetURL()
+	for _, configURL := range localConfigURLs {
+		localURL := ConvertConfigURL(configURL)
+		// search local URL, if it exist in local, update state with remote status
+		if localURL.Name == urlName {
+			if remoteExist {
+				localURL.Status.State = StateTypePushed
+			} else {
+				localURL.Status.State = StateTypeNotPushed
+			}
+			return localURL, nil
+		}
+	}
+
+	if remoteExist {
```

Comment:
also check that the error value from above is nil otherwise the below code will crash

Reply:
oops, yes